pandas/tseries/period.py

```diff
@@ -718,14 +718,15 @@ def __getitem__(self, key):
 
             return PeriodIndex(result, name=self.name, freq=self.freq)
 
-    def _format_native_types(self, na_rep=u('NaT'), **kwargs):
+    def _format_native_types(self, na_rep=u('NaT'), date_format=None, **kwargs):
 
         values = np.array(list(self), dtype=object)
         mask = isnull(self.values)
         values[mask] = na_rep
-
         imask = ~mask
-        values[imask] = np.array([u('%s') % dt for dt in values[imask]])
+
+        formatter = lambda dt: dt.strftime(date_format) if date_format else u('%s') % dt
```

Comment:
put the if on the outside (and lambdas on the inside), then check is only done once